packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -344,6 +350,10 @@ export class AwsCustomResource extends Construct implements iam.IGrantable {
       throw new Error('At least `onCreate`, `onUpdate` or `onDelete` must be specified.');
     }
 
+    if (!props.role && !props.policy) {
+      throw new Error('Either `policy` or `role` must be specified.');
```

Comment:
This makes it sound like they are mutually exclusive. Can you refine this message a little more?

Reply:
This was tricky to get correct, but I've modified the message slightly now to mention both can be used together.